### PR TITLE
Add Tooltip recipe for DataInspector

### PR DESCRIPTION
**Pull request by ffreyer**
# Description

This adds a `tooltip(anchor, text)` recipe which places some text in a box pointing to the anchor position. Here's an example with a few tooltips that I corrently have as a refimg test:

Screenshot from 2022-06-25 15-38-46

I'm planning for this to replace the tooltips created by `DataInspector` which should also make it easier to create persistent tooltips. 

DataInspector:

Screenshot from 2022-06-25 18-06-08

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) because attribute names moved around and some got renamed?

## Checklist

- [x] Added an entry in NEWS.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [x] Added reference image tests for new plotting functions, recipes, visual options, etc.

## TODO

- [x] switch to Tooltip in DataInspector
- ~~add white outline to text in DataInspector tooltips by default... maybe~~
- [x] *figure out heatmap and image tooltips - either set background color and reset afterwards or add some other color indicator* added back the old behavior (cell outline for heatmap, magnified pixel for image)
- [x] fix `contourf` tooltip alignment (and others if there are any)
- [x] fix broken volumeslices
- [x] restore boundingbox/selection indicators (with on/off switch)
- [x] fix duplication problems of some indicators
- [x] add attribute to shift tooltip pointer away from center

### CODE REVIEW

File: src/basic_recipes/tooltip.jl [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        elseif placement in (:above, :up, :top)
+            [
+                Vec2f(0, 0.5w[2]), Vec2f(0, w[2]), w, Vec2f(w[1], 0), 
+                Vec2f(0.5 * (w[1] + s), 0), Vec2f(0.5w[1], -s), 
+                Vec2f(0.5 * (w[1] - s), 0), Vec2f(0), Vec2f(0, 0.5w[2])
+            ]
+        else
+            @error "Tooltip placement $placement invalid. Assuming :above"
+            [
+                VEc2f(0), Vec2f(0, w[2]), w, Vec2f(w[1], 0),
```
Comment (asinghvi17) [inline src/basic_recipes/tooltip.jl:136]:
```suggestion
                Vec2f(0), Vec2f(0, w[2]), w, Vec2f(w[1], 0),

**Comment by ffreyer** [comment]
closes #2092

**Comment by ffreyer** [comment]
Here's a list of changed/moved attributes for `di = DataInspector`. Most of these can be set through keyword arguments in the constructor.

#### Styling attributes

| Changed | old | new |
|:---:| --- | --- |
| x | di.plot.text_padding        | di.plot.textpadding |
| x | di.plot.text_align          | N/A |
|   | di.plot.textcolor           | di.plot.textcolor |
|   | di.plot.textsize            | di.plot.textsize |
|   | di.plot.font                | di.plot.font |
| x | N/A                         | di.plot.justification |
| x | di.plot.background_color    | di.plot.backgroundcolor |
|   | di.plot.outline_color       | di.plot.outline_color |
|   | di.plot.outline_linestyle   | di.plot.outline_linestyle |
| x | di.plot.outline_linewidth   | di.plot.outline_linewidth |
| m | di.plot.indicator_color     | di.attributes.indicator_color |
| m | di.plot.indicator_linewidth | di.attributes.indicator_linewidth |
| m | di.plot.indicator_linestyle | di.attributes.indicator_linestyle |
| x | di.plot.tooltip_offset      | di.attributes.offset (di.plot.offset) |
| x | N/A                         | di.attributes.enable_indicators |
| m | di.plot.range               | di.attributes.range |
| m | di.plot.depth               | di.attributes.depth |

Notes:
- `text_align` has more or less been replaced by `justification`
- `tooltip_offset` has been replaced by `offset::Real`. To set it `di.attributes.offset` should be used. To adjust the final tooltip offset (i.e. for a custom `show_data` method) `dio.plot.offset` should be used.

#### Internal Attributes

| Changed | old | new |
|:---:| --- | --- |
| m | di.plot.enabled             | di.attributes.enabled |
| x | di.plot.tooltip_align       | di.plot.placement |
| x | di.plot._visible            | di.plot.visible |
| x | di.plot._display_text       | di.plot.text |
| x | di.plot._text_position      | di.plot[1] |
| x | di.plot._px_bbox_visible    | di.attributes.indicator_visible |
| x | di.plot._bbox_visible       | di.attributes.indicator_visible |
| x | di.plot._bbox2D             | N/A |
| x | di.plot._bbox3D             | N/A |
| x | di.plot._model              | N/A |
| x | di.plot._position           | N/A |
| x | di.plot._color              | N/A |

Notes:
- `tooltip_align` has been replaced by `placement` which can be :left, :right, :above or :below.
- The attributes from `bbox2D` to `color` were used as reusable Observables for indicators. Instead of those, attributes in `di.temp_plots` are now adjusted directly

**Comment by ffreyer** [comment]
New test image should produce
Screenshot from 2022-07-27 16-39-31

**Review by SimonDanisch — APPROVED** [review]
Looks great! :) I just noticed that the tooltip starts as visible, which then hangs around in the bottom corner... Adding `visible=false` to tooltip in `DataInspector` seems to fix it without staying invisible, so I pushed that change!
I can update the refimages once CI is green

**Comment by ffreyer** [comment]
I moved the changes from #2261 here since the history there got messed up and it's not that big of a change. As such this now closes #1759 and #1902.

**Pull request merged**